### ISSUE TITLE
Does not output xml but html

### ISSUE THREAD

**Issue by powerflow** [Czarto/ShopifyScripts#2]
Hi

I am excited about your code but the output of my collection for fb is still html. I copied and pasted the code into a new file template collection.fb-product-feed.liquid

Can you help me what I am doing wrong?

**Comment by samLozier** [comment]
can you paste your code here? It's important that you clear all css and have this as your first line: 
`{% layout none %}<?xml `version="1.0"?>``

**Comment by reconrad45** [comment]
Thanks sam, I thought the first line needs to be {% section 'collection-template' %} in order to assign the template to collections. I am getting the same error as powerflow in that facebook product catalog says it is an html page. Also, I am still getting the google_product_catalog error even when i use the facebook feed by flexify. In order to troubleshoot, I only added 5 products to the fb-product-feed colleciton. 

In google shopping app, I selected the collection and tried to publish.. received the following error. 
![image](https://user-images.githubusercontent.com/29778420/27691798-25151ce2-5cb3-11e7-8ac3-4bdbf9db72cf.png)

I receive the following error in facebook product catalog fetch (after I re-added {% section 'collection-template' %} at the top of the template because when I took it out it provided a different error that there were no products. 

Without {% section...} code
![image](https://user-images.githubusercontent.com/29778420/27692156-fb915592-5cb3-11e7-8299-22c2c236e3f6.png)

With section code added back in 
![image](https://user-images.githubusercontent.com/29778420/27692222-21e5d600-5cb4-11e7-9d3c-3e16c6e1ec3b.png)

So in summary, facebook catalog doesn't like the format, and google shopping app says that the google_product_discription or catalog isn't provided and facebook gives the same error even when i use facebook feed by flexify. 

Any insight would be appreciated.

**Comment by reconrad45** [comment]
Also if I might add, by adding the {% section 'collection-template' %} at the top of the template I get a hybrid of broken html (images not aligned, etc) and the lower half of the page seems to be a huge block of text of products, description, variants etc, but no xml code present, just standard text & links when going directly to the URL in the browser.

**Comment by reconrad45** [comment]
Of course I am just learning all of this so I am trying to be as clear as possible (probably not working lol)

but here is what I get in the google shopping app even when I point it to a specific collection, (not the facebook feed collection)
![image](https://user-images.githubusercontent.com/29778420/27692921-e9229cac-5cb5-11e7-8298-ab9314a041b7.png)

Regardless of pointing to the a specific collection or the facebook feed collection that I created, I get the same error in google shopping app..

**Comment by samLozier** [comment]
@reconrad45 each product needs to be assigned to a google shopping category so that google and FB (FB uses the same categories) know what type of product you're selling. You can assign the categories during a .csv product upload to shopify, or through the Shopify Google Shopping App. 

Until you have your google taxonomy memorized, I've found that it's easier and more accurate to assign categories through the google shopping app after I've published a product

More on the google taxonomy here:
https://support.google.com/merchants/answer/6324436?hl=en

**Comment by samLozier** [comment]
This is the code that's currently live on my site for google. Once you have categories assigned, you should start seeing good results with this.

```
{% layout none %}<?xml version="1.0"?>
<rss xmlns:g="http://base.google.com/ns/1.0" version="2.0">
{% paginate collection.products by 350 %}
{% assign useSEOtitle = false %}
{% assign useSEOdescription = false %}
{% assign CountryCode = 'US' %}
{% if shop.currency == 'CAD' %}{% assign CountryCode = 'CA' %}{% endif %}
{% assign Color = "" %}
{% assign Size = "" %}
<channel>
<title>{{ shop.name }} {{ collection.title | strip_html | strip_newlines | replace: '&', '&amp;' }}</title>
<link>{{ shop.url }}</link>
<description>{{ collection.description | strip_html | strip_newlines | replace: '&', '&amp;' }}</description>
{% for product in collection.products %} 
  {% assign GoogleProductCategory = product.metafields.google.google_product_type %}
  {% assign Gender = product.metafields.google.gender %}
  {% assign AgeGroup = product.metafields.google.age_group %}
      
  {% for variant in product.variants %}
    {% for option in product.options %}
  	{% if option == 'Color' %}{% capture Color %}{{ variant.options[forloop.index0] }}{% endcapture %}
  	{% elsif option == 'Size' %}{% capture Size %}{{ variant.options[forloop.index0]  }}{% endcapture %}
  	{% endif %}
  	{% endfor %}
    
    {% capture productTitle %}{{ product.vendor }} {{ product.title }} {{ Color }}{% endcapture %}
    {% if useSEOtitle and product.metafields.global.title_tag.size > 0 %}{% assign productTitle = product.metafields.global.title_tag %}{% endif %}
    {% assign productDescrition = product.description %}
    {% if useSEOdescription and product.metafields.global.description_tag.size > 0 %}{% assign productDescrition = product.metafields.global.description_tag %}{% endif %}

<item>
<title>{{ productTitle | replace: '&', '&amp;' }}</title>
<link>{{ shop.url }}{{ variant.url }}</link>
<description>{{ productDescrition | strip_html | strip_newlines | replace: '&', '&amp;' }}</description>
<g:google_product_category>{{ GoogleProductCategory | replace: '&', '&amp;'  }}</g:google_product_category>
<g:item_group_id>shopify_{{ CountryCode }}_{{ product.id }}</g:item_group_id>
<g:id>shopify_{{ CountryCode }}_{{ product.id }}_{{ variant.id }}</g:id>
<g:condition>new</g:condition>
<g:price>{{ variant.price | money_without_currency }} {{ shop.currency }}</g:price>
<g:availability>{% if variant.available %}in stock{% else %}out of stock{% endif %}</g:availability>
<g:image_link>http:{% if variant.image.src %}{{ variant.image.src | product_img_url: 'grande' }}{% else %}{{ product.featured_image.src | product_img_url: 'grande' }}{% endif %}</g:image_link>
<g:gtin>{{ variant.barcode }}</g:gtin>
<g:brand>{{ product.vendor }}</g:brand>
<g:mpn>{{ variant.sku }}</g:mpn>
<g:product_type>{{ product.type }}</g:product_type>
<g:age_group>{{ AgeGroup }}</g:age_group>
{% unless Color == "" %}<g:color>{{ Color | strip_html | strip_newlines | replace: '&', '&amp;' }}</g:color>{% endunless %}
{% unless Size == "" %}<g:size>{{ Size | strip_html | strip_newlines | replace: '&', '&amp;' }}</g:size>{% if product.type == "Apparel" %}<g:size_system>US</g:size_system>{% else %}{% endif %}{% endunless %}
<g:gender>{{ Gender }}</g:gender>
<g:custom_label_0>{{ product.metafields.google.custom_label_0 }}</g:custom_label_0>
<g:custom_label_1>{{ product.metafields.google.custom_label_1 }}</g:custom_label_1>
<g:custom_label_2>{{ product.metafields.google.custom_label_2 }}</g:custom_label_2>
<g:custom_label_3>{{ product.metafields.google.custom_label_3 }}</g:custom_label_3>
<g:custom_label_4>{{ product.metafields.google.custom_label_4 }}</g:custom_label_4>

<g:shipping_weight>{{ variant.weight | weight_with_unit }}</g:shipping_weight>
</item>

{% endfor %}
{% endfor %}
</channel>
</rss>
{% endpaginate %}
```

**Comment by samLozier** [comment]
Final thought: are you assigning this template to your collection? That should happen on the collection page itself. You can select your template towards the bottom right.

**Comment by reconrad45** [comment]
Thanks @samthaman 
"are you assigning this template to your collection?" Yep

![image](https://user-images.githubusercontent.com/29778420/27699789-65be7822-5cc9-11e7-8e03-313dc17d1cea.png)

"You can assign the categories during a .csv product upload to shopify, or through the Shopify Google Shopping App." I have been trying to find how to do this.. as I don't see anything in the google shopping app that lets me edit products and I don't see anything in products that lets me edit using the google shopping app.

**Comment by samLozier** [comment]
In the app, click into a product and then change values as needed here: 
![capture](https://user-images.githubusercontent.com/7992371/27700047-3be6d62e-5cb1-11e7-83a2-6023092d6876.JPG)

**Comment by reconrad45** [comment]
Ah that's it... I assumed that the link in google shopping app was a link to the product page.... hadn't realized that it was a link to adjust the specific google shop settings. Thanks.. now I am getting somewhere lol.

**Comment by reconrad45** [comment]
Progress:
![image](https://user-images.githubusercontent.com/29778420/27700681-4e960b4e-5ccc-11e7-8dc7-4d60a4732589.png)

Makes automation a bit cumbersome needing to go into each and assign a category as well as demands a color variant be created... and if an existing color variant does exist but isn't called straight up "color" it wants you to modify it to color.. i.e. rename "metal color" to just "color.

**Comment by reconrad45** [comment]
Thanks @samthaman for the help, finally getting a grip on how this works.

**Comment by Czarto** [comment]
Seems like this is resolved. Sorry for not getting involved earlier. Thanks @samthaman for troubleshooting. Closing this issue.
